server/src/child.rs

```diff
@@ -88,17 +108,93 @@ struct DebugState {
     last_breakpoint: Option<usize>,
 }
 
+fn attach_pid(pid: u32, tx: SyncSender<DebugMessage>, rx: Receiver<ServerMessage>) -> io::Result<()> {
```

Comment:
Maybe just call this `attach`?

In any case, this looks like the same code from the example you wrote that needs to be moved into the `debug` crate. I.e. this should just call `Child::attach` instead of making winapi calls itself.